### PR TITLE
Turns gloves of the north star functionality into a component.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

title.

## Why It's Good For The Game

The goal of this is to get rid of the Touch hook for gloves. Currently the only other thing that uses this proc is ninja gloves which I will try to remove at a later date.

## Changelog
:cl:
code: Gloves of the North Star functionality is now handled by a component.
/:cl:

### CODE REVIEW

File: code/datums/components/punchcooldown.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
+///Your favourite Jojoke. Used for the gloves of the north star.
```
Comment (MrDoomBringer) [inline code/datums/components/punchcooldown.dm]: >gloves of the north star

>jojoke

🤔

Comment (nemvar) [reply code/datums/components/punchcooldown.dm]: I forgot.

**Comment by GuyonBroadway** [comment]
Touch iirc was a holdover from the old stungloves days. I merely reused it for a meme... Two memes if you count the clown gloves on hippie that force people into help intent if you hug them.

**Comment by nemvar** [comment]
I know that touch is a holdover from stungloves days, doesn't matter that it's not an extremely silly thing to have ingame. We should either use abstract items and/or signals to achieve the same effect.

File: code/datums/components/punchcooldown.dm [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/datum/component/wearertargeting/punchcooldown/Initialize()
+	. = ..()
+	if(. == COMPONENT_INCOMPATIBLE)
+		return
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/changewarcry)
+
+///Called on COMSIG_HUMAN_MELEE_UNARMED_ATTACK. Yells the warcry and and reduces punch cooldown.
+/datum/component/wearertargeting/punchcooldown/proc/reducecooldown(mob/living/carbon/M, atom/target)
+	if(M.a_intent == INTENT_HARM && isliving(target))
+		M.changeNext_move(CLICK_CD_RAPID)
```
Comment (yoyobatty) [inline code/datums/components/punchcooldown.dm]: I wonder if you replace this with a variable to config punching/click delay?

**Comment by zamolxius** [comment]
> I know that touch is a relic from stungloves days, but that doesn't matter. It's an extremely silly thing to have in game. We should either use abstract items and/or signals to achieve the same effect.

ok oranges

**Comment by nemvar** [comment]
Even if stungloves were still in the game Touch would still be shitcode.

**Pull request merged**